### PR TITLE
Reference destructuring from JS3 in React useState lesson

### PR DESCRIPTION
**Pull request by alismith-dev**
## What does this change?

Module: React
Week(s): 2

## Description

As discussed in https://github.com/CodeYourFuture/syllabus/issues/6, we now cover destructuring in JS3. This means that we can reference this in the React module when discussing destructuring `useState`.

[Rendered version](https://github.com/CodeYourFuture/syllabus/blob/react-destructuring/docs/react/week-2/lesson.md)

## Who needs to know about this?

@ChrisOwen101 
@louisechow

### CODE REVIEW

**Comment by 40thieves** [comment]
@ChrisOwen101 can you confirm we're now clear to do this, no cities should be covering React for a while?

File: docs/react/week-2/lesson.md [hunk 1]
```diff
@@ -259,9 +259,17 @@ Finally, let's look at how we get hold of the `count` variable:
 const [count, setCount] = useState(0);
 ```
 
-To fully understand this bit of code, we first have to understand _destructuring_. Let's look at [this blog post by Wes Bos about array destructuring](https://github.com/wesbos/es6-articles/blob/master/19%20-%20Destructing%20Arrays.md).
+If you remember back to [JS Core 3 Week 3](https://syllabus.codeyourfuture.io/js-core-3/week-3/lesson#destructuring-syntax), we covered _destructuring_. As a reminder, you can assign variables to parts of an array with destructuring:
```
Comment (ChrisOwen101) [inline docs/react/week-2/lesson.md]: Could this be tweaked to be a relative link?

**Review by ChrisOwen101 — APPROVED** [review]

File: docs/react/week-2/lesson.md (hunk 1 above)
Comment (alismith-dev) [reply docs/react/week-2/lesson.md]: 👍 Done in https://github.com/CodeYourFuture/syllabus/pull/136/commits/b364f37d48f5b1919a5228e09a950a61689dd2d5

**Review by illicitonion — APPROVED** [review]
Looks good!

**Pull request merged**